examples/kmeans/kmeans.py

```diff
@@ -0,0 +1,93 @@
+import numpy as np
+
+import cupy
+
+
+class KMeans(object):
+    def __init__(self, n_clusters=2, max_iter=10):
+        self.n_clusters = n_clusters
+        self.max_iter = max_iter
+        self.centers = None
+        self.xp = None
+        self.pred = None
+
+    def fit(self, X):
+        self.xp = cupy.get_array_module(X)
+        self.pred = self.xp.zeros(len(X))
+        X_indexes = np.arange(len(X))
+        np.random.shuffle(X_indexes)
+        initial_centroid_indexes = X_indexes[:self.n_clusters]
+        self.centers = X[initial_centroid_indexes]
+        data_num = X.shape[0]
+        data_dim = X.shape[1]
+
+        for _ in range(self.max_iter):
+            # calculate distances and label
+            distances = self.xp.zeros((data_num, self.n_clusters),
+                                      dtype=self.xp.float64)
+            if self.xp == np:
+                for i in range(self.n_clusters):
+                    distances[:, i] = self.xp.sum((X - self.centers[i]) ** 2,
```

Comment:
`distances` is used to calculate only the IDs of clusters like below. I think we do not have to calculate accurate distances.
```
new_pred = xp.argmin(distances, axis=1)
```